### PR TITLE
fix: Calling video not streamed when enabling camera on preview screen (WPB-7114)

### PR DESCRIPTION
**Pull request by ohassine**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/WPB-7114" title="WPB-7114" target="_blank">BugWPB-7114</a>  [Android] When video is enabled on precall screen, video is not streamed during 1:1 and group call
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [x] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

Calling video not streamed when enabling camera on preview screen of an incoming call.

### Causes (Optional)

TLDR: A race condition.
Establishing an incoming calls takes some times an important time, causing to call the function of video stream before the call get established.

### Solutions

I moved the logic of sending video feed to OngoingCallViewModel, this way we are sure `setVideoSendState` will be called only when the call get established.

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

- Start a call to an android device.
- On Android enable your camera and accept the call
- You should still see your camera on, and the other person see your video feed.

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [3661](https://10.10.124.134/job/AR-build-pipeline/3661/) **failed**.

**Comment by AndroidBob** [comment]
Build [3662](https://10.10.124.134/job/AR-build-pipeline/3662/) **failed**.

**Comment by AndroidBob** [comment]
Build [3663](https://10.10.124.134/job/AR-build-pipeline/3663/) **failed**.

**Comment by AndroidBob** [comment]
Build [3664](https://10.10.124.134/job/AR-build-pipeline/3664/) **failed**.

**Review by alexandreferris — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [3667](https://10.10.124.134/job/AR-build-pipeline/3667/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.7.0(9198046)(9198046)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.7.0(9198046)(9198046)-staging-compat.apk)

**Review by MohamadJaara — APPROVED** [review]
Please remember to cherry-pick this change into RC

**Comment by AndroidBob** [comment]
Build [3719](https://10.10.124.134/job/AR-build-pipeline/3719/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.7.0(9239319)(9239319)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.7.0(9239319)(9239319)-staging-compat.apk)
